libs/api/domains/health-directorate/src/lib/health-directorate.service.ts

```diff
@@ -24,84 +20,78 @@ export class HealthDirectorateService {
         return {
           id: item.id,
           name: item.name,
         }
       }) ?? []
 
-    return exceptions
+    return limitations
   }
 
-  async updateDonorStatus(auth: Auth, input: DonorStatusInput): Promise<void> {
-    await this.organDonationApi
-      .updateOrganDonation(auth, {
-        exceptionComment: input.exceptionComment ?? '',
-        isDonor: input.isDonor,
-        exceptions: input.exceptions ?? [],
-        registrationDate: new Date(),
-      })
-      .then(() => {
-        return true
-      })
-      .catch(() => {
-        return false
-      })
+  async updateDonorStatus(auth: Auth, input: DonorInput): Promise<void> {
+    return await this.organDonationApi.updateOrganDonation(auth, {
+      isDonor: input.isDonor,
+      exceptions: input.organLimitations ?? [],
+    })
   }
 
   /* Vaccinations */
-  async getVaccinations(
-    auth: Auth,
-    locale: Locale,
-  ): Promise<Array<Vaccinations>> {
-    const data = await this.vaccinationApi.getVaccinationDiseaseDetail(auth, {
-      locale,
-    })
-    const vaccinations: Array<Vaccinations> =
+  async getVaccinations(auth: Auth): Promise<Vaccinations> {
+    const data = await this.vaccinationApi.getVaccinationDiseaseDetail(auth)
+    const vaccinations: Array<Vaccination> =
       data?.map((item) => {
         return {
-          diseaseId: item.diseaseId,
-          diseaseName: item.diseaseName,
-          diseaseDescription: item.diseaseDescription,
-          vaccinationStatus: item.vaccinationStatus,
-          vaccinationsStatusName: item.vaccinationStatusName,
-          lastVaccinationDate: item.lastVaccinationDate,
-          vaccinations:
-            item.vaccinations?.map((vaccination: VaccinationDto) => {
+          id: item.diseaseId,
+          name: item.diseaseName,
+          description: item.diseaseDescription,
+          isFeatured: item.isFeatured,
+          status: item.vaccinationStatus,
+          statusName: item.vaccinationStatusName,
+          statusColor: item.vaccinationStatusColor,
+          lastVaccinationDate: item.lastVaccinationDate ?? null,
+          comments: item.comments,
+          vaccinationsInfo: item.vaccinations?.map(
+            (vaccination: VaccinationDto) => {
               return {
                 id: vaccination.id,
-                nationalId: vaccination.nationalId,
-                code: vaccination.code,
-                vaccinationDate: vaccination.vaccinationDate,
-                vaccinationsAge: vaccination.vaccinationAge,
-                generalComment: vaccination.generalComment.toString(),
+                name: vaccination.vaccineCodeDescriptionShort,
+                date: vaccination.vaccinationDate,
+                age: vaccination.vaccinationAge,
+                url: vaccination.vaccineUrl,
+                comment: vaccination.generalComment,
                 rejected: vaccination.rejected,
-              }
-            }) ?? [],
+              } as Info
```

Comment:
Does this circumvent the type checking?

Reply:
Not sure what you mean?

Reply:
i find it's slightly sketchy using the `as` keyword when mapping, since it sometimes introduces mapping bugs which are hard to notice. If you're 100% confident the object are the same then it's fine.

Reply:
Ah I didn't even see the "as" until you said it, I agree with you, don't know why I used it 🐒